### ISSUE TITLE
Webpack - injecting third party lib

### ISSUE THREAD

**Issue by amirmog** [angular-fullstack/generator-angular-fullstack#2237]
This is not an issue, just a simple question.
Hope this helps other ppl who have the same question.

It's my first time trying to use angular-fullstack.
I installed and everything seems to be running fine.

Now I simply want to be able to use this third-party lib in my angular-fullstack project.
http://angular-ui.github.io/angular-google-maps/#!/use

How can I add it to the the angular-fullstack so that the lib is injected into the main controller and can be used everywhere?

Maybe it's ue to my lack of familiarity with the Webpack and the way it is setup here but any help would be so much appreciated.

Thanks,

**Comment by aliirz** [comment]
+1  I need to know this as well.

**Comment by thedewpoint** [comment]
If the library supports Commons then you just need to npm install --save libraryname. Then if you go into the web pack file you will see where angularjs etc is imported. It should be an array of strings. Just go ahead and add the name of your library there. Then in the controller of your project , at the top type import library from 'libraryname'. If it's exported as an angular module you will have to include it in your main module as a dependency.

**Comment by thedewpoint** [comment]
Commonjs not Commons, using my phone sorry.

**Comment by aliirz** [comment]
@thedewpoint  its not working for me. I am trying to install and use https://github.com/720kb/angular-datepicker in my angular-fullstack app. I have installed it via npm and added it to webpack.make.js file like this:

```
if(TEST) {
        config.entry = {};
    } else {
        config.entry = {
            app: './client/app/app.js',
            polyfills: './client/polyfills.js',
            vendor: [
                'angular',
                'angular-animate',
                'angular-aria',
                'angular-cookies',
                'angular-resource',

                'angular-sanitize',
                'angular-socket-io',
                'angular-ui-bootstrap',
                'angular-ui-router',
                'lodash',
                '720kb.datepicker'
            ]
        };
    }
```

I get the following error in browser console:

```
Error: Cannot find module "720kb.datepicker"
```

very new to webpack and missing bower at this point :(

**Comment by amirmog** [comment]
👍

**Comment by thedewpoint** [comment]
@aliirz I believe you need to add angularjs-datepicker to your web pack file, not the angular module name

**Comment by Koslun** [comment]
You need to also import or require it in one of the files loaded by webpack as well. Meaning in either your entry file i.e. `./client/app/app.js` directly or by something else loaded by the entry file. You could for example load a dependency in some controller which is then loaded by importing a module such as the controller in `account/login.js` which is exported in the `account` module.

```javascript
import libName from 'libName';
```

If you instead want to load an angular module and thus get it injected in your app module so you can use it as you would expect in the rest of your angular app you would probably do best to simply load it in the entry file. For that you need to either import, or require the module to get the javascript and if you additionally also need css you need to import that separately. 

Import examples:

```javascript
import 'angular-socket-io';

const ngRoute = require('angular-route');
const ngMessages = require('angular-messages');

import 'angular-material/angular-material.css';
const ngMaterial = require('angular-material');
```

To get it loaded into your angular module you need to further get the names of each angular module into the angular dependency array. Depending on your module you might be able to simply insert the variable into the array or have to write it manually. The above modules could be imported like this:

```javascript
angular.module('myApp', [
    //... other modules

    'btford.socket-io',

    ngRoute,
    ngMessages,

    ngMaterial,
  ])
```

**Comment by thedewpoint** [comment]
You guys still stuck? Can this be closed?

**Comment by Awk34** [comment]
@Koslun 's answer is correct. @aliirz if you wanted to include that lib in the `vendor.bundle.js` file, you'd use the npm name 'angularjs-datepicker' instead of '720kb.datepicker'.

**Comment by amirmog** [comment]
Thank you all for the contribution.
I managed to get it working as you all mentioned:

Added the import into webpack.build.js
ie.

                'ngmap'

Then inside the main.controller.js imported the module
import NgMap from 'ngmap';

Then added it to the controctor of the main.controller.js
  constructor($http, $scope, socket, $timeout, NgMap) {/*...*/}

Though that said, injecting into the controller is not always needed. It's up to the library that you are adding.
ie. Angular Material Design only needed to be imported:
import ngMaterial from 'angular-material';

Thanks all.
Cheers.
a.
